xtable-core/src/main/java/org/apache/xtable/avro/AvroSchemaConverter.java

```diff
@@ -435,6 +435,11 @@ private Schema fromInternalSchema(InternalSchema internalSchema, String currentP
             Schema.createFixed(
                 internalSchema.getName(), internalSchema.getComment(), null, fixedSize),
             internalSchema);
+      case UUID:
+        // Convert to fixed 16-byte array since Avro does not support UUID natively
+        return finalizeSchema(
+            Schema.createFixed(internalSchema.getName(), internalSchema.getComment(), null, 16),
+            internalSchema);
```

Comment:
This converter is currently used exclusively for Hudi target conversions. Are there any other components or processes that need to handle the UUID internal type for Hudi?

Reply:
Nothing else is coming to mind. For partitions, we only support the year/month/day/hour and identity transforms from Iceberg current so those partition functions would just pass through the value if I'm remembering correctly.